Calculator.rb

```diff
@@ -0,0 +1,141 @@
+supported_ope = ["add", "+", "subtract", "-", "multiply", "*", "divide", "/",
+  "power of", "^", "modulo", "%"]
+
+$users_numbers = []
+$ope = ""
+def ask_number_input(fun_input)
+  auth_char = ["1","2","3","4","5","6","7","8","9","0",".","-"]
+  ask_number = "yes"
+
+
+  while ask_number == "yes"
+    puts "\nGive me a number:"
+    num = gets.chomp
+    if num == fun_input
+      $users_numbers << fun_input
+      ask_number = "no"
+    else
+      num_chars = num.split("")
+      forbiden_char_found = "no"
+      dot_count = 0
+      dash_count = 0
+
+      num_chars.each do |char|
+        unless auth_char.include? char
+          forbiden_char_found = "yes"
+        end
+        if char == "."
+          dot_count += 1
+        elsif char == "-"
+          dash_count += 1
+        end
+      end
+      if forbiden_char_found == "yes"
+        puts "\nYou can only enter numbers."
+      elsif $ope == "%"  && $users_numbers.length == 1 && num == "0"
+        puts "\nOops, can't divide by 0."
+      elsif $ope == "/"  && $users_numbers.length == 1 && num == "0"
```

Comment:
You're missing the possibility that the divide operation could be "divide" and not "/"